File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    void printRightView(List<TreeNode> ans, TreeNode root, int level){
        if(root==null){
            return;
        }
        if(ans.size()==level){
            ans.add(root);
        }
        printRightView(ans, root.right, level+1);
        printRightView(ans, root.left, level+1);
    }
    public List<Integer> rightSideView(TreeNode root) {
        List<TreeNode> ans = new ArrayList<>();
        printRightView(ans, root, 0);
        List<Integer> result = new ArrayList<>();
        for(TreeNode temp: ans){
            result.add(temp.val);
        }
        return result;
    }
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void printRightView(vector<TreeNode*>& ans, TreeNode* root, int level){
        if(root==NULL){
            return;
        }
        if(ans.size()==level){
            ans.push_back(root);
        }
        printRightView(ans, root->right, level+1);
        printRightView(ans, root->left, level+1);
    }
    vector<int> rightSideView(TreeNode* root) {
        vector<TreeNode*> ans;
        printRightView(ans, root, 0);
        vector<int> result;
        for(TreeNode* temp: ans){
            result.push_back(temp->val);
        }
        return result;
    }
};